pkg/webui/components/key-value-map/index.js

```diff
@@ -0,0 +1,90 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import { Container, Col, Row } from 'react-grid-system'
+import bind from 'autobind-decorator'
+import classnames from 'classnames'
+
+import PropTypes from '../../lib/prop-types'
+import Button from '../button'
+import Entry from './entry'
+
+@bind
+class KeyValueMap extends React.PureComponent {
+
+  handleEntryChange (index, newValues) {
+    this.props.onChange(this.props.value.map((kv, i) => (
+      index !== i ? kv : { ...kv, ...newValues }
+    )))
+  }
+
+  removeEntry (index) {
+    this.props.onChange(this.props.value.filter((_, i) => (i !== index)) || [])
+  }
+
+  addEmptyEntry () {
+    this.props.onChange([ ...this.props.value, { key: '', value: '' }])
+  }
+
+  render () {
+    const {
+      className,
+      name,
+    } = this.props
+
+    return (
+      <Container className={classnames(className, this.props)} style={{ paddingLeft: '0px' }}>
+        {this.props.value && this.props.value.map((value, index) => (
+          <Entry
+            key={`${name}[${index}]`}
+            name={name}
+            value={value}
+            keyPlaceholder={this.props.keyPlaceholder}
+            valuePlaceholder={this.props.valuePlaceholder}
+            index={index}
+            onRemoveButtonClick={this.removeEntry}
+            onChange={this.handleEntryChange}
+          />
+        ))}
+        <Row style={{ paddingTop: '4px' }}>
+          <Col>
+            <Button
```

Comment:
Let's make the button more descriptive by adding an `Add header entry` message as well.